sdk/python/feast/client.py

```diff
@@ -146,6 +147,15 @@ def _core_service(self):
             self._core_service_stub = CoreServiceStub(channel)
         return self._core_service_stub
 
+    @property
+    def _use_object_store_registry(self) -> bool:
```

Comment:
I think it's important to have a clean migration path for teams that are adopting the object store registry. If we don't implement the Core reader, then my suggested approach is something like
1. Add object store registry support to SDK (what you are doing)
2. Add a conditional flag like `FALL_BACK_TO_CORE_REGISTRY`. That flag is `True` by default, meaning that  clients will connect to and use the core registry if no registry file is found in the object store during read methods. During create/update operations I think we will have to always create the file, but most of the time the apply methods are controlled by a single team, not by many clients.
3. Teams would migrate by upgrading all client SDKs to the latest version
4. The core registry owners would then shut down their core registry and migrate their configuration to the object store
5. All clients in (3) would then see this new registry file and start using it.
6. The core registry can then be removed entirely.

Reply:
cc @oavdeev

Reply:
can we do something simpler like:
1. Team that owns Feast installation creates object store by using the (soon-coming) migration script
2. All clients upgraded to use the object store; either no applies are allowed during this period or any applies done to the core registry must be reflected manually to the object store registry
3. Team removes core registry

Reply:
It's kind of a hairy upgrade path. What if one team (a Feast user, not owners of system) says that they have a critical production system and they can't upgrade their client for at least 2-3 weeks? That means you need to take Feast Core down for 2-3 weeks while you wait for that team to upgrade. It's not always possible to get everyone to upgrade at the same point in time, so naturally different teams will upgrade at different points in time. The easiest is to provide some kind of middle ground.